supabase/migrations/20260715213729_app_preview_api_key_role.sql

```
@@ -0,0 +1,1201 @@
+
+-- Preview-channel grants are children of their app_preview binding. The foreign
+-- key revokes them when the parent is removed; RBAC also checks the parent is
+-- still active so expiration changes take effect immediately.
+ALTER TABLE public.role_bindings
+  ADD COLUMN IF NOT EXISTS parent_binding_id uuid;
+
+ALTER TABLE public.role_bindings
+  ADD CONSTRAINT role_bindings_parent_binding_id_fkey
+  FOREIGN KEY (parent_binding_id)
+  REFERENCES public.role_bindings(id)
+  ON DELETE CASCADE;
```

SQLFluff (4.2.2)

[error] ADD CONSTRAINT ... FOREIGN KEY should use NOT VALID to avoid locking the table while validating existing rows.

(PG01)